app/controllers/barbeque/sns_subscriptions_controller.rb

```diff
@@ -44,6 +46,12 @@ def destroy
   private
 
   def fetch_sns_topic_arns
-    Barbeque::SNSSubscriptionService.sns_client.list_topics.topics.map(&:topic_arn)
+    if Barbeque.config.sns_regions.empty?
+      Aws::SNS::Client.new.list_topics.topics.map(&:topic_arn)
```

Comment:
Aws::SNS::Clinet should be memoized because Aws::SNS::Client.new retrieves credentials each time when ECS task role or EC2 instance profile is available.

Reply:
06d3a6c ✨

Reply:
2a14998 (revised commit comment 💦 )